streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -218,6 +218,10 @@ public void initializeIfNeeded() {
      */
     @Override
     public void completeRestoration() {
+        if (state() == State.RUNNING) {
+            return;
+        }
+
         if (state() == State.RESTORING) {
             initializeMetadata();
             initializeTopology();
```

Comment:
github won't let me leave a comment below this line, but can we use the `"Illegal state"`/`"Unknown state"` improvement in this method as well?